artemis-protocols/artemis-openwire-protocol/src/main/java/org/apache/activemq/artemis/core/protocol/openwire/OpenWireMessageConverter.java

```diff
@@ -196,7 +200,21 @@ private OpenWireMessageConverter() {
 
       final ActiveMQDestination replyTo = messageSend.getReplyTo();
       if (replyTo != null) {
-         putMsgReplyTo(replyTo, marshaller, coreMessage);
+         if (replyTo instanceof TemporaryQueue) {
+            MessageUtil.setJMSReplyTo(coreMessage, org.apache.activemq.artemis.jms.client.ActiveMQDestination.TEMP_QUEUE_QUALIFED_PREFIX + (((TemporaryQueue) replyTo).getQueueName()));
```

Comment:
Same query i put on the AMQP one but also relevant i think here, but why dont we use the same logic when you transform a JMS Message for foreign Messages on producer send (which you have to handle as part of TCK)

Reply:
What Query you referring to? This is about conversions.. I'm not sure how the TCK applies here.

Reply:
you resolved it, just unresolved it.

Reply:
https://github.com/apache/activemq-artemis/blob/4ca3a971baeb97623384ddc17893f7ebaf8cace1/artemis-jms-client/src/main/java/org/apache/activemq/artemis/jms/client/ActiveMQMessageProducer.java#L416

Reply:
and for doing it for OpenWire, we could levy its implementation, and like wise for AMQP can use qpid-jms varient.

Reply:
This is dealing directly with the properties here, I'm not going to implement ServerQueue, ServerTopic, ServerTemporaryTopic, ServerTemporaryQueue just to set the property on this case and still not solve it completely.

Reply:
it would mean duplciation regardless. The client JMS implementation is not available at the server's. This is only needed when converting this property between protocols.

Reply:
Just to be clear this is more a general query, and question for longer term maintenance of this stuff... its not needed to be addressed as part of this PR, just this PR raises the question.